homeassistant/components/acmeda/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""The Rollease Acmeda Automate integration."""
+import asyncio
+import ipaddress
+
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+from .hub import PulseHub
+
+CONF_HUBS = "hubs"
+
+HUB_CONFIG_SCHEMA = vol.Schema(
+    {
+        # Validate as IP address and then convert back to a string.
+        vol.Required(CONF_HOST): vol.All(ipaddress.ip_address, cv.string),
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_HUBS): vol.All(
+                    cv.ensure_list, [vol.All(HUB_CONFIG_SCHEMA)],
+                )
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+PLATFORMS = ["cover", "sensor"]
+
+
+async def async_setup(hass: core.HomeAssistant, config: dict):
+    """Set up the Rollease Acmeda Automate component."""
+    conf = config.get(DOMAIN)
+    if conf is None:
+        conf = {}
+
+    hass.data[DOMAIN] = {}
+
+    # User has configured hubs
+    if CONF_HUBS not in conf:
+        return True
+
+    hubs = conf[CONF_HUBS]
+
+    configured_hubs = {
+        entry.data["host"] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+    for hub_conf in hubs:
+        host = hub_conf[CONF_HOST]
+
+        if host in configured_hubs:
```

Comment:
We don't need this check. We do it in the config flow too.

Reply:
Removed